tests/unit/test_synapse_workload.py

```diff
@@ -22,6 +22,94 @@
 from .conftest import TEST_SERVER_NAME
 
 
+def test_enable_ip_range_whitelist_success(harness: Harness, monkeypatch: pytest.MonkeyPatch):
+    """
+    arrange: set mock container with file.
+    act: update ip_range_whitelist config and call enable_ip_range_whitelist.
+    assert: new configuration file is pushed and ip_range_whitelist is enabled.
+    """
+    config_content = """
+    listeners:
+        - type: http
+          port: 8080
+          bind_addresses:
+            - "::"
+    """
+    text_io_mock = io.StringIO(config_content)
+    pull_mock = Mock(return_value=text_io_mock)
+    push_mock = MagicMock()
+    container_mock = MagicMock()
+    monkeypatch.setattr(container_mock, "pull", pull_mock)
+    monkeypatch.setattr(container_mock, "push", push_mock)
+
+    expected_first_domain = "foo1"
+    expected_second_domain = "foo2"
+    harness.update_config(
+        {"ip_range_whitelist": f"{expected_first_domain},{expected_second_domain}"}
```

Comment:
This is somewhat limited testing, should we add a [pytest parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html) with a bunch of valid options?

Reply:
I changed my approach actually. Instead of adding unit test for this, I created an integration test that check if the configuration is validated or not.

Reply:
I'll close this comment since is outdated so new comments can be made in this new version but feel free to re-open it.